adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java

```diff
@@ -0,0 +1,163 @@
+package uk.gov.justice.raml.jms.core;
+
+import static java.lang.String.format;
+import static org.apache.commons.io.FileUtils.write;
+import static uk.gov.justice.raml.jms.core.TemplateRenderer.render;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.raml.model.Action;
+import org.raml.model.ActionType;
+import org.raml.model.Raml;
+import org.raml.model.Resource;
+
+import uk.gov.justice.raml.core.Generator;
+import uk.gov.justice.raml.core.GeneratorConfig;
+import uk.gov.justice.services.core.annotation.Component;
+
+/**
+ * Generates JMS endpoint classes out of RAML object
+ *
+ */
+public class JmsEndpointGenerator implements Generator {
+
+    private static final String UTF_8 = "UTF-8";
+    private static final String TEMPLATE_LOADING_ERROR = "Failed to load template resource JmsListenerTemplate.tpm";
+    private static final String ACTIONS_EMPTY_ERROR = "No actions to process";
+    private static final String OUTPUT_FILE_GENERATION_ERROR = "Failed to create output file for %s";
+    private static final String FILENAME_POSTFIX = "JmsListener.java";
+    private static final String JMS_TEMPLATE_RESOURCE = "JmsListenerTemplate.tpm";
+    private static final Pattern MEDIA_TYPE_PATTERN = Pattern.compile("(application/vnd.)(\\S+)(\\+\\S+)");
+
+    /**
+     * Generates JMS endpoint classes out of RAML object
+     * 
+     * @param raml
+     *            - the RAML object
+     * @param generatorConfig
+     *            - contains package of generated sources, as well as source and
+     *            destination folders
+     */
+    @Override
+    public void run(final Raml raml, final GeneratorConfig configuration) {
+        final File outputDirectory = createOutputDirectories(configuration);
```

Comment:
We already create and clear the output directory in the plugin - we should standardize on that approach, I think.